tensorrt_llm/_torch/models/modeling_deepseekv3.py

```diff
@@ -1108,19 +1131,22 @@ def forward(
         input_ids: torch.IntTensor = None,
         position_ids: Optional[torch.IntTensor] = None,
         inputs_embeds: Optional[torch.FloatTensor] = None,
-        spec_metadata: Optional[MTPSpecMetadata] = None,
+        spec_metadata: Optional[SpecMetadata] = None,
         return_context_logits: bool = False,
         **kwargs,
     ) -> torch.Tensor:
+        #TODO Izzy check this
         attn_metadata.num_generations_per_batch = self.model_nextn + 1
         hidden_states = self.model(
             input_ids=input_ids,
             attn_metadata=attn_metadata,
             position_ids=position_ids,
             inputs_embeds=inputs_embeds,
+            spec_metadata=spec_metadata,
         )
 
         if spec_metadata and spec_metadata.spec_dec_mode.is_mtp():
+            # TODO Merge API with EagleWorker in modeling_speculative.py
```

Comment:
We should do this now that there are multiple possible 1-model paths. Otherwise I think it's a bit too hard to follow

Reply:
Per our discussion offline we can do this on a followup. Let's open a JIRA though so it doesn't get swept under the rug